### DISCUSSION TITLE
Can I build a dashboard that will enable users to connect to an external API and manage that data?

### DISCUSSION THREAD

**Discussion by easaw** [Budibase/budibase#9187] [Help]
_(I don't have discord, so thought I'd post here)_

Hi, I'd like to build a system on Budibase, I have both the Cloud version and self-hosted, I've looked into about 15 other tools (including Retool, Zoho, Bubble, Adalo and AppSmith) and most of them have missing features, or are restricted, or charge per (external) user.

Here's what I'd like to do:
- Have a login/ registration page that is _not_ tied to Budibase users but has separate data per company.
- Have separate database tables (or entirely separate DB) per client.
- The registration creates an 'account' for a company which has a custom field for a company ID to assign users.
- Allow clients to connect their own data via external API (login and connect via OAuth2) to manage that data
- Invite and manage users who will signup on mobile via Firebase, who are also linked to the company ID, who count towards a "user limit" (again, nothing to do with Budibase users)
- Charge a monthly fee (I have the Stripe plugins), that also restricts the number of users.
- Track numbers of users and frequency of logins and display info and user logs on a dashboard.

_Notes:_
_- I'll be using n8n to do automations, probably via webhooks and return data to Budubase_
_- I don't want to have to manually create company accounts (i.e. database per company)_
_- The external data connection is a 'requirement' of this service, and I would need a page to allow the company owner to do a callback to connect their OAuth2 account_

If anyone has any advice, especially about connecting an external API per client, and a login service that isn't tied to Budibase, that would be great.

Thank you.

**Comment by melohagan** [comment]
Hey @easaw 

Thanks for having a look at Budibase!

You can use OAuth2 to allow users to login, however at the moment they would be tied to Budibase users, and you cannot specify your own Identity Provider.

Separate tables per client, and managing users per client is not something Budibase is directly designed for, i.e. it's not a tool for building SaaS platforms.

Tracking user activities through audit logs is yet to be built, but will hopefully be released this year.
